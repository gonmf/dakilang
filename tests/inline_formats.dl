number(N, 'float', Res) :- set(N + 3.14, Res).
number(N, 'decimal', Res) :- set(N + 119, Res).
number(N, 'octal', Res) :- set(N + 0170, Res).
number(N, 'hexadecimal', Res) :- set(N + 0x79, Res).
number(N, 'binary', Res) :- set(N + 0b1111010, Res).

number(5, Format, Res)?

listing
