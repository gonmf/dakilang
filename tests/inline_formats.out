> number(N, 'float', Res) :- set(N + 3.14, Res).
> number(N, 'decimal', Res) :- set(N + 119, Res).
> number(N, 'octal', Res) :- set(N + 0170, Res).
> number(N, 'hexadecimal', Res) :- set(N + 0x79, Res).
> number(N, 'binary', Res) :- set(N + 0b1111010, Res).
>
> number(5, Format, Res)?
Format = 'float'
Res = 8.14

Format = 'decimal'
Res = 124

Format = 'octal'
Res = 125

Format = 'hexadecimal'
Res = 126

Format = 'binary'
Res = 127

>
> listing
0: number(N, 'float', Res) :- eval(N, '$0+3.14', $0), set($0, Res).
1: number(N, 'decimal', Res) :- eval(N, '$0+119', $0), set($0, Res).
2: number(N, 'octal', Res) :- eval(N, '$0+120', $0), set($0, Res).
3: number(N, 'hexadecimal', Res) :- eval(N, '$0+121', $0), set($0, Res).
4: number(N, 'binary', Res) :- eval(N, '$0+122', $0), set($0, Res).

